llmfoundry/models/hf/hf_causal_lm.py

```diff
@@ -59,32 +57,51 @@ class ComposerHFCausalLM(HuggingFaceModelWithFSDP):
         tokenizer (PreTrainedTokenizer): The tokenizer that the model will use.
     """
 
-    def __init__(self, om_model_config: DictConfig,
-                 tokenizer: PreTrainedTokenizerBase):
+    def __init__(
+        self,
+        tokenizer: PreTrainedTokenizerBase,
+        pretrained_model_name_or_path: str,
+        pretrained: Optional[bool] = True,
+        pretrained_lora_id_or_path: Optional[str] = None,
+        trust_remote_code: bool = True,
+        use_auth_token: bool = False,
+        use_flash_attention_2: bool = False,
+        load_in_8bit: bool = False,
+        init_device: str = 'cpu',
+        config_overrides: Optional[Dict[str, Any]] = None,
+        peft_config: Optional[Dict[str, Any]] = None,
+        use_train_metrics: bool = True,
+        additional_train_metrics: Optional[List] = None,
+        additional_eval_metrics: Optional[List] = None,
+        # ignored args
+        name: Optional[str] = None,
```

Comment:
This constructor has many callsites and it seemed better to accept and ignore a `name`, which we may use for logging or validation than to pop name from the kwargs at the callsites. I'm open to doing the latter if you think it would be better